airflow/models/dagbag.py

```diff
@@ -640,8 +640,10 @@ def _sync_perm_for_dag(self, dag, session: Session = None):
         from airflow.security.permissions import DAG_ACTIONS, resource_name_for_dag
         from airflow.www.fab_security.sqla.models import Action, Permission, Resource
 
+        root_dag_id = dag.parent_dag.dag_id if dag.parent_dag else None
```

Comment:
Does this need to account for nested DAGs? (i.e. a subdag whose parent is also a subdag. Is this a thing?)

Reply:
I had to go look the documentation. From what I can see, a subdag can't have a subdag as the parent.
cc: @ashb @kaxil